hop/subscribe.py

```diff
@@ -6,28 +6,61 @@
 
 import argparse
 import json
+import warnings
 
 from . import cli
 from .io import Stream
-from .models import GCNCircular
+from .models import GCNCircular, VOEvent
 
 
-def print_gcn(gcn_dict, json_dump=False):
-    """Parse a gcn dictionary and print to stdout.
+def print_msg(msg, json_dump=False):
+    """Check the format of a message obtained from an ADC stream, use it to instantiate
+    a data model corresponding to that format, and print the message.
 
     Args:
-      gcn_dict:  the dictionary object containing a GCN-formatted message
+      msg:       the raw message from an ADC stream
+      json_dump: boolean indicating whether to print msg as raw json
 
     Returns:
       None
-
     """
+
+    # check for msg format using standard-specific flags
+    voevent_flag = 'ivorn'
+    gcncir_flag = 'GCN CIRCULAR'
+
+    # VOEvent:
+    if voevent_flag in msg:
+        gcn = VOEvent(**msg)
```

Comment:
Please rename the `gcn` variable now that it can be a VOEvent message.

Reply:
I believe the nomenclature has `gcn` as the parent message class, where `gcn circular` and `gcn notice` (which includes `VOEvent`) are instantiations of `gcn`.

Reply:
I'm inclined to agree with renaming the `gcn` variable as well, maybe something like `model`(?) to reflect the return description in the docstring, "dataclass model object for the raw message".